xtask/src/commands/release.rs

```diff
@@ -0,0 +1,545 @@
+use anyhow::{anyhow, Result};
+use cargo_metadata::MetadataCommand;
+use itertools::Itertools;
+use octorust::types::{
+    IssuesCreateMilestoneRequest, IssuesListMilestonesSort, IssuesListState, IssuesUpdateRequest,
+    Milestone, Order, PullsCreateRequest, State, TitleOneOf,
+};
+use octorust::Client;
+use structopt::StructOpt;
+use tap::TapFallible;
+use walkdir::WalkDir;
+use xtask::*;
+
+#[derive(Debug, StructOpt)]
+pub enum Command {
+    /// Prepare a new release
+    Prepare(Prepare),
+}
+
+impl Command {
+    pub fn run(&self) -> Result<()> {
+        match self {
+            Command::Prepare(command) => command.run(),
+        }
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Prepare {
+    /// Release from the current branch rather than creating a new one.
+    #[structopt(long)]
+    current_branch: bool,
+
+    /// Skip the license check
+    #[structopt(long)]
+    skip_license_ckeck: bool,
+
+    /// Dry run, don't commit the changes and create the PR.
+    #[structopt(long)]
+    dry_run: bool,
+
+    /// The new version that is being created OR to bump (major|minor|patch|current).
+    version: Option<String>,
```

Comment:
Should be an enum IMHO

Reply:
[b171f48](https://github.com/apollographql/router/pull/2202/commits/b171f4804c439be1074409a4e5ea6a3a7a413fd8)